(#1095) Remove code from ctor

baudoliver7 opened the pull request:
For #1095

- Change code in ctor to `Sticky` initialization
- Remove lazy loading evaluation in `RqMtFake`

Commit:
Update jcabi-parent to latest version #958

Commit:
Trigger github actions #958

Commit:
Fix class unfound #958

Commit:
Remove code from ctor RqMtBase ##1095

andreoss commented:
@baudoliver7 The tests are failing

andreoss commented on pom.xml:
```diff
@@ -532,4 +528,4 @@ SOFTWARE.
     </profile>
   </profiles>
-</project>
+</project>
```
@baudoliver7 I suggest reverting this part, if xcop doesn't complain than xml formating is fine

andreoss commented on src/main/java/org/takes/rq/multipart/RqMtBase.java:
```diff
@@ -125,27 +127,24 @@
-            Math.min(8192, this.stream.available())
-        );
-        this.map = this.requests(req);
+        this.sstream = new Sticky<>(() -> new RqLengthAware(req).body());
```
@baudoliver7 Is it possible to use `RqOnce` on the ctor parameter instead of three Stickies? 
It seems also that some fields could be moved inside of `request` method, such as `sbuffer`

yegor256 commented:
@baudoliver7 what's up with this one?

yegor256 commented:
Please fix the merge conflicts so this pull request can be merged.

baudoliver7 committed:
Merge branch 'master' into remove_code_from_ctor

baudoliver7 committed:
fix: fix issues introduced by merging

baudoliver7 committed:
fix: fix pom.xml

baudoliver7 committed:
Update multipart tests for lazy (code-free) ctor

RqMtBase no longer parses in its constructor, so invalid multipart
input now surfaces as UncheckedIOException at first use of names()
instead of a checked IOException at construction. Trigger evaluation
via names() and assert UncheckedIOException in the affected tests.

baudoliver7 committed:
Restore code-free RqMtFake ctor to keep API backward-compatible

A bad merge had regressed RqMtFake to an eager `RqMultipart` field
whose ctor called `new RqMtBase(...)` directly, forcing it to declare
`throws IOException`. Revapi flagged this as a checked-exception added
to a public constructor. Restore master's lazy `Scalar<RqMultipart>`
field so the ctor is code-free and no longer throws.

baudoliver7 commented:
@yegor256 It is good now :)
